io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -724,4 +730,29 @@ private List<String> writeToFile(
 		return filesWritten;
 	}
 
+	/**
+	 * 
+	 * @param filePath Path of the file to be written with just one line of the
+	 *            encoding
+	 * @throws IOException exception in any.
+	 */
+	public void writeCodePageFile(String filePath) throws IOException {
+		File file = new File(filePath);
+		FileWriter fileWriter = new FileWriter(file);
+		try {
+			fileWriter.write(getCharset() != null ? getCharset().toString()
```

Comment:
Now that you use the writer setting to document the encoding in the CPG file, you also need to make sure that the writer does write the Shapefile with this encoding. May assumption would be that this can be done [similar to in the reader](https://github.com/halestudio/hale/blob/37d0458c0868ccf5ff0adf4cb44bb9b7fb07776a/io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/reader/internal/ShapeInstanceReader.java#L105).

Reply:
> Now that you use the writer setting to document the encoding in the CPG file, you also need to make sure that the writer does write the Shapefile with this encoding.

I would expect this to require something like

```
newDataStore.setCharset(getCharset());
```

to be done [here](https://github.com/halestudio/hale/blob/68175508ef152681bfb0ac4f9432074f1cf7701c/io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java#L463).

Reply:
I see. In the [reader is done ](https://github.com/halestudio/hale/blob/68175508ef152681bfb0ac4f9432074f1cf7701c/io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/reader/internal/ShapeInstanceReader.java#L105) properly but not in the writer.  I'll update the code.